core/src/wallet/ripple/explorers/NodeRippleLikeBlockchainExplorer.cpp

```diff
@@ -0,0 +1,282 @@
+/*
+ *
+ * NodeRippleLikeBlockchainExplorer
+ *
+ * Created by El Khalil Bellakrid on 09/01/2019.
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+
+#include "NodeRippleLikeBlockchainExplorer.h"
+#include <api/RippleConfigurationDefaults.hpp>
+#include <api/Configuration.hpp>
+#include <rapidjson/document.h>
+namespace ledger {
+    namespace core {
+        NodeRippleLikeBlockchainExplorer::NodeRippleLikeBlockchainExplorer(
+                const std::shared_ptr<api::ExecutionContext> &context,
+                const std::shared_ptr<HttpClient> &http,
+                const api::RippleLikeNetworkParameters &parameters,
+                const std::shared_ptr<api::DynamicObject> &configuration) :
+                DedicatedContext(context),
+                RippleLikeBlockchainExplorer(configuration, {api::Configuration::BLOCKCHAIN_EXPLORER_API_ENDPOINT}) {
+            _http = http;
+            _parameters = parameters;
+        }
+
+
+        Future<std::shared_ptr<BigInt>>
+        NodeRippleLikeBlockchainExplorer::getBalance(const std::vector<RippleLikeKeychain::Address> &addresses) {
+            //TODO: multiple accounts balances ?
```

Comment:
What do you mean?

Reply:
I removed it, those were thoughts at the beginning of implementation :trollface: